apps/files_trashbin/lib/Sabre/TrashbinPlugin.php

```
@@ -129,4 +132,40 @@ public function httpGet(RequestInterface $request, ResponseInterface $response):
 			$response->addHeader('Content-Disposition', 'attachment; filename="' . $node->getFilename() . '"');
 		}
 	}
+
+	/**
+	 * Check if a user has available space before attempting to
+	 * restore from trashbin unless they have unlimited quota.
+	 *
+	 * @param string $sourcePath
+	 * @param string $destinationPath
+	 * @return bool
+	 */
+	public function beforeMove(string $sourcePath, string $destinationPath): bool {
+		try {
+			$node = $this->server->tree->getNodeForPath($sourcePath);
+			$destinationNode = $this->server->tree->getNodeForPath(dirname($destinationPath));
+		} catch (\Exception $e) {
+			return true;
```

You should only catch \Sabre\DAV\Exception\NotFound which you can silently ignore. If you ever need to catch a broad set of exceptions make sure to always log the exceptions. Otherwise it can be really hard to debug if you only have the logs and no step-debugger.